### PR TITLE
[unify] Fix unify job performance

### PR DESCRIPTION
**Pull request by jjmerchante**
This PR updates the way unify job fetches all the individuals from the database. Instead of retrieving identities from each individual, fetch all the identities at once.

### CODE REVIEW

File: sortinghat/core/recommendations/matching.py [hunk 1]
```diff
@@ -94,13 +94,20 @@ def _get_identities(uuid):
...
+            input_set.update(identities)
+    else:
+        identities = Identity.objects.select_related('individual').all()
         input_set.update(identities)
+        for identity in identities:
+            if identity.individual.mk in aliases:
+                aliases[identity.individual.mk].append(identity.uuid)
+            else:
+                aliases[identity.individual.mk] = [identity.uuid]
+        source_uuids = aliases.keys()
```
Comment (sduenas) [inline sortinghat/core/recommendations/matching.py:110]: What about using `defaultdict`? https://docs.python.org/3/library/collections.html#defaultdict-examples

File: tests/test_jobs.py [hunk 2]
```diff
@@ -594,8 +594,8 @@ def test_recommend_matches_all_individuals(self):
         }
 
         recommendations_expected = {
-            self.john_smith.individual.mk: [self.jsmith.individual.mk, self.js_alt.individual.mk],
-            self.jsmith.individual.mk: [self.js_alt.individual.mk],
+            self.js_alt.individual.mk: [self.jsmith.individual.mk, self.john_smith.individual.mk],
```
Comment (sduenas) [inline tests/test_jobs.py:597]: Why does this test require the change?

Comment (jjmerchante) [reply tests/test_jobs.py:597]: The order of the recommendations changed...

**Review by sduenas — CHANGES_REQUESTED** [review]
LGTM but please, add a changelog file with something like:

```
Performance improved for recommendations and merging jobs

The performance of the matching and merging algorithms that
are used on these jobs have been considerably improved.
These jobs will be 4 times faster than on the previous version.
```

**Pull request merged**